client/driver/raw_exec.go

```diff
@@ -75,6 +73,48 @@ func (d *RawExecDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandl
 	}
 	taskLocal := filepath.Join(taskDir, allocdir.TaskLocal)
 
+	// Get the command to be ran, or if omitted, download an artifact for
+	// execution. Currently a supplied command takes precedence, and an artifact
+	// is only downloaded if no command is supplied
+	command, ok := task.Config["command"]
+	if !ok || command == "" {
+		return nil, fmt.Errorf("missing command for exec driver")
+	}
+
+	source, ok := task.Config["artifact_source"]
+	if ok && source != "" {
+		// Proceed to download an artifact to be executed.
+		// We use go-getter to support a variety of protocols, but need to change
+		// file permissions of the resulted download to be executable
+
+		// Create a location to download the artifact.
+		destDir := filepath.Join(taskDir, allocdir.TaskLocal)
+
+		artifactName := path.Base(source)
+		artifactFile := filepath.Join(destDir, artifactName)
+		if err := getter.GetFile(artifactFile, source); err != nil {
+			return nil, fmt.Errorf("[Err] driver.Exec: Error downloading source for Exec driver: %s", err)
+		}
+
+		// Add execution permissions to the newly downloaded artifact
+		if runtime.GOOS != "windows" {
+			if err := syscall.Chmod(artifactFile, 0655); err != nil {
+				log.Printf("[Err] driver.Exec: Error making artifact executable: %s", err)
+			}
+		}
+	}
+
+	// expand NOMAD_TASK_DIR
+	ntdMap := make(map[string]string)
```

Comment:
Missed this in my commit. You don't have to do this as NOMAD_TASK_DIR is already in the map returned from TaskEnvironmentVariables(). So just move that down under envVars and use that map. You can follow the example in the executors.